metricflow-semantics/metricflow_semantics/model/semantics/metric_lookup.py

```diff
@@ -80,6 +83,7 @@ def linkable_elements_for_no_metrics_query(
             without_any_of=frozen_without_any_of,
         )
 
+    @functools.cache
     def linkable_elements_for_metrics(
         self,
         metric_references: Sequence[MetricReference],
```

Comment:
Oof Sequence isn't counted as immutable. Should we update this to Tuple or does the typechecker catch it if we pass in a non-hashable type?

Reply:
The type checker catches it! It fails if the type isn't hashable.